### PR TITLE
DEPR: Index.format

### LINKED ISSUE #55413: DEPR: Index.format?
Does anyone actually use this?  Internally we use it for some html/latex rendering but we don't use most of the keywords so it would be simpler to refactor to use a private method for those.

I've been trying to unravel the various EA formatting methods to try to use the same patterns everywhere, align keywords, etc.  Doing this for Index.format wouldn't be _that_ hard, but it would be easier to just deprecate it if it isn't used externally.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #55413 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by attack68 — APPROVED** [review]
lgtm in my files of interest

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1383,6 +1391,31 @@ def format(
 
         return self._format_with_header(header=header, na_rep=na_rep)
 
+    _default_na_rep = "NaN"
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:1394]: Is this correct if the data is a masked array with NA?

Comment (jbrockmendel) [reply pandas/core/indexes/base.py:1394]: ATM Index.format and DTI.format have different defaults for na_rep.  Putting this here keeps the same behavior for _format_flat while avoiding the need for overrding the whole method in DTI.

Looks like na_rep is ignored inside format_with_header anyway

Comment (mroeschke) [reply pandas/core/indexes/base.py:1394]: > Looks like na_rep is ignored inside format_with_header anyway

That's probably not good, but I guess that can be tackled elsewhere

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**